kiwi/config/functions.sh

```diff
@@ -545,13 +545,16 @@ function baseVagrantSetup {
     # insert the default insecure ssh key from here:
     # https://github.com/hashicorp/vagrant/blob/master/keys/vagrant.pub
     mkdir -p /home/vagrant/.ssh/
+    chmod 0700 /home/vagrant/.ssh/
     echo "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkTkyrtvp9eWW6A8YVr+kz4TjGYe7gHzIw+niNltGEFHzD8+v1I2YJ6oXevct1YeS0o9HZyN1Q9qgCgzUFtdOKLv6IedplqoPkcmF0aYet2PkEDo3MlTBckFXPITAMzF8dJSIFo9D8HfdOV0IAdx4O7PtixWKn5y2hMNG0zQPyUecp4pzC6kivAIhyfHilFR61RGL+GPXQ2MWZWFYbAGjyiYJnAmCP3NOTd0jMZEnDkbUvxhMmBYSdETk1rRgm+R4LOzFUGaHqHDLKLX+FIPKcF96hrucXzcWyLbIbEgE98OHlnVYCzRdK8jlqm8tehUc9c9WhQ== vagrant insecure public key" > /home/vagrant/.ssh/authorized_keys
     chmod 0600 /home/vagrant/.ssh/authorized_keys
     chown -R vagrant:vagrant /home/vagrant/
 
+    SSHD_CONFIG=/etc/ssh/sshd_config.d/99-vagrant.conf
```

Comment:
You'll need to test for the directory, since it doesn't exist in RHEL 7, RHEL 8, SLE 12, or SLE 15. When it doesn't exist, fallback to the old behavor of extending `/etc/ssh/sshd_config`.

Reply:
Right, that makes sense. I've changed the behavior accordingly and added a test on CentOS7 just to be sure, which surfaced another bug in this function as well (the custom settings need to get prepended, not appended…).